### ISSUE TITLE
`no-fallthrough` ignores `never` function return types

### ISSUE THREAD

**Issue by jsejcksn** [denoland/deno_lint#804]
### Lint Name
[`no-fallthrough`](https://lint.deno.land/#no-fallthrough)

### Code Snippet

```ts
const int = Math.floor(Math.random() * 4);

function printAndExit (msg: string, code = 0): never {
  console[code === 0 ? 'log' : 'error'](msg);
  Deno.exit(code);
}

function main () {
  switch (int) {
    case 0: {
      console.log('👍');
      Deno.exit(0);
    }
    case 1: {
      console.log('👍');
      return;
    }
    case 2: {
      printAndExit('👍');
    }
    default: {
      printAndExit('👎', 1);
    }
  }
}

main();

```

### Expected Result

```
$ deno lint lint-example.ts
Checked 1 file
```

### Actual Result

```
$ deno lint lint-example.ts
(no-fallthrough) Fallthrough is not allowed
    case 0: {
    ^^^^^^^^^^
      console.log('👍');
^^^^^^^^^^^^^^^^^^^^^^^^^^^
      Deno.exit(0);
^^^^^^^^^^^^^^^^^^^^
    }
^^^^^
    at /Users/jesse/example/lint-example.ts:10:4

    hint: Add `break` or comment `/* falls through */` to your case statement
    help: for further information visit https://lint.deno.land/#no-fallthrough

(no-fallthrough) Fallthrough is not allowed
    case 2: {
    ^^^^^^^^^^
      printAndExit('👍');
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    }
^^^^^
    at /Users/jesse/example/lint-example.ts:18:4

    hint: Add `break` or comment `/* falls through */` to your case statement
    help: for further information visit https://lint.deno.land/#no-fallthrough

Found 2 problems
Checked 1 file
```

### Additional Info

Regarding code flow analysis, it seems that using the `return` keyword is interpreted as a terminal statement, but that invoking functions whose return types are `never` are not interpreted as terminal.

### Version

```
$ deno --version
deno 1.13.1 (release, x86_64-unknown-linux-gnu)
v8 9.3.345.11
typescript 4.3.5
```

**Comment by magurotuna** [comment]
Duplicate of #782 
To deal with all the possible situations, type analysis is required, but unfortunately we don't have that. Maybe we should try to find a solution for simple cases like the example you described.
